### PR TITLE
Gjoranv/prometheus handler

### PR DESCRIPTION
**Pull request by gjoranv**
Based on https://github.com/vespa-engine/vespa/pull/10192

@yj-jtakagi:
Some changes from your original PR: 

- We decided to implement a separate rest api with base path `prometheus/v1´.
- I removed the service name as part of the metric name and instead add a new label/dimension called `vespa_service`.

I added your github username as author in the class headers. Feel free to send PRs with corrections, test improvements etc.

### CODE REVIEW

**Comment by gjoranv** [comment]
Test failure in node-repository is irrelevant:
[ERROR]   LoadBalancerExpirerTest.expire_inactive:72 Inactive load balancer removed

File: metrics-proxy/src/main/java/ai/vespa/metricsproxy/http/TextResponse.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    private final byte[] data;
+
+    public TextResponse(int code, String data) {
+        super(code);
+        this.data = data.getBytes(Charset.forName(DEFAULT_CHARACTER_ENCODING));
+    }
+
+    @Override
+    public void render(OutputStream outputStream) throws IOException {
+        outputStream.write(data);
```
Comment (bjorncs) [inline metrics-proxy/src/main/java/ai/vespa/metricsproxy/http/TextResponse.java]: Default content-type is plaintext?

File: metrics-proxy/src/main/java/ai/vespa/metricsproxy/metric/model/prometheus/PrometheusModel.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        return metricFamilySamplesIterator.next();
+    }
+
+    public String serialize() {
+        var writer = new StringWriter();
+        try {
+            TextFormat.write004(writer, this);
+        } catch (IOException e) {
+            log.log(Level.WARNING, "Got exception when rendering metrics:", e);
+            throw new PrometheusRenderingException("Could not render metrics. Check the log for details.");
```
Comment (bjorncs) [inline metrics-proxy/src/main/java/ai/vespa/metricsproxy/metric/model/prometheus/PrometheusModel.java]: Consider including the cause in exception

**Review by bjorncs — APPROVED** [review]

File: metrics-proxy/src/main/java/ai/vespa/metricsproxy/http/TextResponse.java (hunk 1 above)
Comment (gjoranv) [reply metrics-proxy/src/main/java/ai/vespa/metricsproxy/http/TextResponse.java]: Yes, default mimetype is text/plain in HttpResponse. We can probably never change this.

File: metrics-proxy/src/main/java/ai/vespa/metricsproxy/metric/model/prometheus/PrometheusModel.java (hunk 2 above)
Comment (gjoranv) [reply metrics-proxy/src/main/java/ai/vespa/metricsproxy/metric/model/prometheus/PrometheusModel.java]: It's included in the log, but we don't want to show it in the response.

**Pull request merged**